### PR TITLE
update style

### PR DESCRIPTION
**Pull request by dirk-thomas**
Connect to ~~ament/uncrustify#20~~. Connect to ament/uncrustify#21.

Ready for review.

### CODE REVIEW

File: rcl/src/rcl/client.c [hunk 1]
```diff
@@ -211,9 +211,7 @@ rcl_client_get_service_name(const rcl_client_t * client)
   return client->impl->rmw_handle->service_name;
 }
 
-/* *INDENT-OFF* */
-#define _client_get_options(client) &client->impl->options;
-/* *INDENT-ON* */
+#define _client_get_options(client) & client->impl->options;
```
Comment (wjwwood) [inline rcl/src/rcl/client.c]: This was on purpose. I don't think the new version handles this correctly still.

Comment (wjwwood) [reply rcl/src/rcl/client.c]: In each of the cases below as well where a reference is used in the macro.

File: rcl/test/memory_tools/test_memory_tools.cpp [hunk 2]
```diff
@@ -20,22 +20,22 @@
  */
 TEST(TestMemoryTools, test_allocation_checking_tools) {
   size_t unexpected_mallocs = 0;
-  auto on_unexpected_malloc = (
-    [&unexpected_mallocs]() {
+  auto on_unexpected_malloc = [&unexpected_mallocs]()
+    {
       unexpected_mallocs++;
-    });
+    };
```
Comment (wjwwood) [inline rcl/test/memory_tools/test_memory_tools.cpp]: These look very strange, would this not work instead?

```c++
  auto on_unexpected_malloc = [&unexpected_mallocs]() {
    unexpected_mallocs++;
  };
```

Or:

```c++
  auto on_unexpected_malloc =
    [&unexpected_mallocs]() {
      unexpected_mallocs++;
    };
```

For me, this should look like an initializer list, so like this:

```c++
foo = {
  'a',
  'b',
}
```

And not:

```c++
foo =
  {
    'a',
    'b',
  };
```

Comment (dirk-thomas) [reply rcl/test/memory_tools/test_memory_tools.cpp]: Please try if the suggested alternative passes `uncrustify`. I really don't mind how it looks like.

File: rcl/src/rcl/client.c (hunk 1 above)
Comment (dirk-thomas) [reply rcl/src/rcl/client.c]: > I don't think the new version handles this correctly still.

Can you clarify what you mean with this? The current patch passes fine with the current `uncrustify` version. That is why I removed the markers around the line.

File: rcl/test/memory_tools/test_memory_tools.cpp (hunk 2 above)
Comment (dirk-thomas) [reply rcl/test/memory_tools/test_memory_tools.cpp]: Please try if the suggested alternative passes `uncrustify` (I think I recall it doesn't). I really don't mind how it looks like.

Comment (dirk-thomas) [reply rcl/test/memory_tools/test_memory_tools.cpp]: The first options doesn't pass. The second one does. Updated in https://github.com/ros2/rcl/pull/215/commits/dfd8ba25fcd9d440505e03b40161e264a9fd41e3#diff-b4937ca0f22682aded6f4aeaa9d0392bR23

Comment (wjwwood) [reply rcl/test/memory_tools/test_memory_tools.cpp]: Thanks, lgtm.

File: rcl/src/rcl/client.c (hunk 1 above)
Comment (wjwwood) [reply rcl/src/rcl/client.c]: It's not very important, but there shouldn't be a space between the `&` and the code, that's why the `INDENT-OFF` was there before.

Comment (dirk-thomas) [reply rcl/src/rcl/client.c]: If you prefer that I can revert that change.

Comment (wjwwood) [reply rcl/src/rcl/client.c]: You can or not, like I said not that important.

**Pull request merged**